client/signup/config/flows.js

```diff
@@ -65,7 +65,7 @@ function getSiteDestination( dependencies ) {
 function getRedirectDestination( dependencies ) {
 	if (
 		dependencies.oauth2_redirect &&
-		dependencies.oauth2_redirect.startsWith( 'https://public-api.wordpress.com' )
+		new URL( dependencies.oauth2_redirect ).host === 'public-api.wordpress.com'
```

Comment:
Can we also handle the potential type errors (https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#exceptions) on invalid urls?